### PR TITLE
Simplify the autogenerated JSON schema closer to a usable form (Infra)

### PR DESCRIPTION
**Pull request by mz2**
## Description

Cleans up the previously autogenerated JSON schema to a closer to usable form.

This is probably the most boring PR you ever read, but it probably makes sense to at least spot check commit by commit. 

## Documentation

`submission-schema/README.md` updated.

## Tests

Submission schema is neither used for generating client code nor validated against yet, so no tests either.

### CODE REVIEW

**Comment by mz2** [comment]
> Unless I'm missing something, the --rename is redundant with --new-name. And I added some cleaner alternatives below (not blocking the landing).

Yeah, you're right, this was just a GPT assisted turd I used to assist in a fast cleanup. I'll clean it up into a neat form.

File: submission-schema/manipulate-schema.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+# A tool to collapse or rename a type definition in a JSON schema.
+
+import json
+import argparse
+
+
+def modify_definition(schema, definition_key, replacement, rename, new_name):
+    if rename and definition_key in schema["definitions"]:
+        schema["definitions"][new_name] = schema["definitions"][definition_key]
+        del schema["definitions"][definition_key]
```
[the remark this answers is not in the document]
Comment (mz2) [reply submission-schema/manipulate-schema.py:12]: I'll keep to the current definition here, applied the rest (readability improvement here is arguable).

**Comment by mz2** [comment]
> Unless I'm missing something, the --rename is redundant with --new-name.
> And I added some cleaner alternatives below (not blocking the landing).

Yep, that was redundant indeed, addressed with updates just now and applied all but one of your suggestions overall.

**Review by kissiel — APPROVED** [review]
Thanks for the amendments, +1

File: submission-schema/manipulate-schema.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    parser.add_argument(
+        "-n",
+        "--new-name",
+        type=str,
+        default=None,
+        help="New name for the definition if renaming.",
+    )
+
+    args = parser.parse_args()
+    if args.rename and args.replacement:
```
[the remark this answers is not in the document]
Comment (mz2) [reply submission-schema/manipulate-schema.py]: Agreed, I removed this and simplified the arg handling.

**Pull request merged**